src/types/presets/aws-amplify.ts

```diff
@@ -0,0 +1,161 @@
+   *
+   * @example `"entrypoint.js"`
+   */
+  entrypoint: string;
+}
+
+export type AmplifyRouteTarget =
+  | { kind: "Static"; cacheControl?: string }
+  | { kind: "ImageOptimization"; cacheControl?: string }
+  | {
+      kind: "Compute";
+      /**
+       * A string that indicates the name of the sub-directory in the given deployment bundle that
+       * contains the primitive's executable code. Valid and required only for the Compute primitive.
+       * The value here should point to one of the compute resources present in the given
+       * deployment bundle. The only supported value for this field is default.
+       */
+      src: string;
+    };
+
+export type AmplifyRoute = {
+  /**
+   * The path defines a glob pattern that matches incoming request paths (excluding querystring).
+   * The first match in a given list of rules determines which routing rule is applied to the incoming request.
+   * Only the following wilcard characters are supported as far as pattern matching is concerned: `*` (matches 0 or more characters)
+   *
+   * _Note_: The "/*" pattern is called a catch-all pattern and will match all incoming requests.
+   * It is special because fallback routing is only supported for catch-all routes.
+   *
+   */
+  path: string;
+
+  /**
+   * An object that dictates the target to route the matched request to.
+   */
+  target: AmplifyRouteTarget;
+
+  /** An object that dictates the target to fallback to if the original target returns a 404. */
+  fallback?: AmplifyRouteTarget;
+};
+
+export type AmplifyImageSettings = {
+  /** Array of supported image widths */
+  sizes: number[];
+
+  /**
+   * Array of allowed external domains that can use Image Optimization.
+   * Leave empty for only allowing the deployment domain to use Image Optimization.
+   */
+  domains: string[];
+
+  /**
+   * Array of allowed external patterns that can use Image Optimization.
+   * Similar to `domains` but provides more control with RegExp.
+   */
+  remotePatterns: {
+    /** The protocol of the allowed remote pattern. Can be `http` or `https`. */
+    protocol?: "http" | "https";
+    /**
+     * The hostname of the allowed remote pattern.
+     * Can be literal or wildcard. Single `*` matches a single subdomain.
+     *  Double `**` matches any number of subdomains.
+     * We will disallow blanket wildcards of `**` with nothing else.
+     */
+    hostname: string;
+    /** The port of the allowed remote pattern. */
+    port?: string;
+    /** The pathname of the allowed remote pattern. */
+    pathname?: string;
+  }[];
+
+  /** Array of allowed output image formats. */
+  formats?: (
+    | "image/avif"
+    | "image/webp"
+    | "image/gif"
+    | "image/png"
+    | "image/jpeg"
+  )[];
+
+  /** Cache duration (in seconds) for the optimized images. */
+  minimumCacheTTL?: number;
+
+  /** Allow SVG input image URLs. This is disabled by default for security purposes. */
+  dangerouslyAllowSVG?: boolean;
```

Comment:
All these 3 fields are required:
- `formats` (Can be empty array)
- `minimumCacheTTL` (60 is a reasonable default)
- `dangerouslyAllowSVG` (`false` is a reasonable default)